tests/framework/web/ResponseTest.php

```diff
@@ -311,4 +311,21 @@ public function testSendFileWithInvalidCharactersInFileName()
 
         $this->assertSame("attachment; filename=\"test_test.txt\"; filename*=utf-8''test%7Ftest.txt", $response->headers['content-disposition']);
     }
+
+    public function testSameSiteCookie()
+    {
+        $response = new Response();
+        $response->cookies->add(new \yii\web\Cookie([
+            'name'     => 'test',
+            'value'    => 'testValue',
+            'sameSite' => \yii\web\Cookie::SAME_SITE_STRICT,
+        ]));
+
+        ob_start();
+        $response->send();
+        $content = ob_get_clean();
+
+        // Only way to test is that it doesn't create any errors
```

Comment:
Would it write anything to headers_list()?

Reply:
No headers are handled separately. setcookie is a php function that gets handled by webserver. Unles you have a way to make a test do a legitimate request I don't see a way. I did test the code on our application and worked. I'll leave it up to you to resolve.

Reply:
OK.